index.js

```diff
@@ -14,44 +13,89 @@ function createResolver(config, options) {
     resolve: resolve,
   };
 
-  var lastKey;
 
-  function resolve(key) {
-    if (typeof key !== 'string') {
+  // Keep constants separately
+  var constants = {};
+
+  function resolveConstant(key) {
+    if (constants.hasOwnProperty(key)) {
+      return constants[key];
+    }
+
+    var definition = config[key];
+    // Ignore options that are not defined
+    if (!definition) {
       return;
     }
 
-    var appliedArgs = slice.call(arguments, 1);
+    var option = options[key];
+
+    if (!!option || options.hasOwnProperty(key)) {
+      if (typeof option !== 'function') {
```

Comment:
If you invert this, you can just return early and reduce a bunch of this indentation.  It would also group things nicely at the bottom of the function and avoid your `// Fall through` scenario.  If you simplify the above line, you could group the early return into a single `if` statement.